### ISSUE TITLE
docs(stepper): add better documentation and examples

### ISSUE THREAD

**Issue by mmalerba** [angular/material2#10381]
- Need lots more examples (currently only one)
- Better explanation of interaction between various state flags (editable, completed, etc) - which combo leads to which icon

**Comment by WizardPC** [comment]
Can be linked to #10365 ?

**Comment by raygervais** [comment]
I'd like to try this as my first contribution to this project, but I need a few pointers such as what would make the documentation ideal, any things I should know about the stepper which is perhaps not in the current documentation, etc. Thanks!

**Comment by mmalerba** [comment]
@raygervais A good first step would be to convert the existing text-only examples to embedded examples: https://material.angular.io/components/stepper/overview

You could also look over the various components that make up the stepper (https://github.com/angular/material2/tree/master/src/cdk/stepper  https://github.com/angular/material2/tree/master/src/lib/stepper) and make sure that the docs covers the various `@Input()` and `@Output()` options they support.

**Comment by mmalerba** [comment]
Yep. Testing the examples is kind of a pain though, you have to clone our docs repo as well https://github.com/mmalerba/material.angular.io and run `npm run fetch-local` to import the examples from your local copy of the material2 repo. If you have trouble getting that to work just test them out in a StackBlitz and I can verify it in the docs site before approving.

**Comment by raygervais** [comment]
Noted, I have the `fetch-local` working so far, but when I attempt to view it locally via `nom start`, it fails to compile. Furthermore what is StackBlitz? 😨

**Comment by raygervais** [comment]
@mmalerba, I've created a WIP PR so you can tell me what else we should have or what needs to be changed. Looking forward to this contribution helping to improve documentation 😄
